programs/drift/src/math/margin.rs

```diff
@@ -240,6 +240,39 @@ pub fn calculate_perp_position_value_and_pnl(
     ))
 }
 
+pub fn calculate_user_safest_position_tiers(
+    user: &User,
+    perp_market_map: &PerpMarketMap,
+    spot_market_map: &SpotMarketMap,
+) -> DriftResult<(AssetTier, ContractTier)> {
+    let mut safest_tier_spot_liablity: AssetTier = AssetTier::default();
+    let mut safest_tier_perp_liablity: ContractTier = ContractTier::default();
+
+    for spot_position in user.spot_positions.iter() {
+        if spot_position.scaled_balance == 0 && spot_position.open_orders == 0
```

Comment:
i think you can check !is_available() instead of looking at scaled_balance and open orders

same goes for perp position

Reply:
using `is_available()`